subsys/caf/modules/shell.c

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright (c) 2023 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <zephyr/shell/shell.h>
+#include <app_event_manager.h>
+#include <caf/events/button_event.h>
+
+static int set_button_handler(const struct shell *shell, size_t argc, char **argv)
+{
+	if (argc < 3) {
+		shell_error(shell, "Missing required parameter(s)");
+		return -EINVAL;
+	}
+
+	struct button_event *event = new_button_event();
+
+	event->key_id = (uint16_t)atoi(argv[1]);
+	event->pressed = *argv[2] != '0';
```

Comment:
Maybe we can define it simply as `y` or `n` and reject other values here (you could also ensure that parameter length is 1 here)? This seems more straightforward for booleans than using numbers.
btw. currently pressed would also be `false` e.g. for value of `01` here which may not be very intuitive.

@jayteemo what do you think about it ?

Reply:
`y` or `n` seems fine

Reply:
`y` or `n` and rejecting anything else seems reasonable to me as well